frontend/pages/shelter/[shelterId]/pet/[petId].tsx

```diff
@@ -128,10 +78,12 @@ export default function PetDetails() {
 				image: "https://via.placeholder.com/22",
 			},
 		];
-		setPetData(pd as Animal.Attributes);
-		setPetAdopters(adopterData);
+		setPetAdopters(mockAdopterData);
+		fetchPetDetail();
 	}, [petId]);
 
+	if (loading) return <Container>LOADINGGGG</Container>;
```

Comment:
if you want, can use [`Spin`](https://ant.design/components/spin/) as a loader and [`Result`](https://ant.design/components/result/) to display errors. can ref [this page](https://github.com/bettersg/PawScore/blob/master/frontend/components/adopter/ViewPets/index.tsx#L141-L161)